lib/mongoid/association/referenced/has_many/enumerable.rb

```diff
@@ -216,6 +216,21 @@ def empty?
               end
             end
 
+            # Does the enumerable have any? Will determine if there exists at least
+            # one element in the enumerable, whether or not it is _loaded.
+            #
+            # @example Doess the enumerable have any?
+            #   enumerable.any?
+            #
+            # @return [ true, false ] If the enumerable has any elements.
+            def any? 
+              if _loaded?
+                in_memory.count > 0
+              else
+                _unloaded.exists? || _added.count > 0
```

Comment:
This is looking great so far! I have two remaining questions:
1. Under what circumstances does this `_added.count` case get executed?
2. If that happened in a transaction, would it cause an error?

Apologies if you're already testing that case, but I just want to make sure that's covered.

Reply:
Good catch! I just added a test like this...

Reply:
Nice, thank you!